Assignment2 @BenagainB

BenagainB opened the pull request:
Initial work for Assignment2

tremorris1999 commented:
I will review.

tremorris1999 reviewed (changes requested):
## Instructions
Note: Custom CSS can only be used where specified.

1. Create a custom component to show available words ❌
- Add a question mark key to the keyboard. This represents a placeholder for any character ✅
- Create/extend a word list component with a method called validWords that takes a string with question marks and returns an array of valid words ❌
- Create unit tests for the validWords method ❌
- Display the number of valid words based on the entered letters ❌
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ❌
- When a word is clicked, the word should be entered as the guess but not submitted ❌
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ❌

2. Update the App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ❌✅ (Torn because this is a home button not a back button)

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ❌
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ❌
- Add gradient to guesses blocks and keys (with CSS) ❌
- Add drop shadow to keys (No CSS) ❌
- Make the app responsive (No CSS) ❌

5. Include a link to your Azure App in your PR comments.


## Summary
Overall it looks pretty good. Missing some parts but looks good where done. Be sure to finish up!

mpennock5 reviewed (commented):
A couple things that I found:

- The color schemes do not have an option to go back to the default, only to change it from the default. 
- The about page link throws error 404. I think it is because you have the about page as a component instead of a page? I am not sure but I would check that out. I do see that you specify the path in your layout so maybe it is an error on my environment but I would look into making that a page instead of component. 
- There are two menu icons in the top right of the app bar.
- The available words isn't yet implemented. We used a regex passed into a filter function to return a new list in one of the scripts file. 
Everything else is looking pretty good!

BenagainB commented:
@mpennock5 Thanks! I added a default color option and Hewr moved out about page to pages instead of components. We also removed the duplicate menu icon. Thank you for the feedback with leads on how to find/fix them.

BenagainB commented:
updated deployment location:
https://nice-river-0685dad0f.1.azurestaticapps.net/

mmwoodfo commented on wordle-web/components/settings-dialog.vue:
```diff
@@ -39,6 +39,12 @@
                   <v-list-item @click="purpleTheme">
                     <v-list-item-title> Purple </v-list-item-title>
                   </v-list-item>
+                  <v-list-item @click="fallTheme">
```
It may be nice to have a way to switch back to the default theme.

mmwoodfo commented on wordle-web/components/settings-dialog.vue:
```diff
@@ -79,9 +80,34 @@ export default class SettingsDialog extends Vue {
+      error: colors.orange.accent3,
+      success: colors.brown.lighten4,
+    }
+    this.$vuetify.theme.themes.dark = fallTheme
+    this.$vuetify.theme.themes.light = fallTheme
```
You may consider wrapping this into its own function called something like 'setTheme' that takes in that object and sets it. That way you aren't writing this twice. Not a big deal though. Good job here.

mmwoodfo commented on wordle-web/components/keyboard.vue:
```diff
@@ -3,8 +3,9 @@
-          <v-btn
+          <v-btn 
             :color="letterColor(char)"
+            :elevation="hover ? 24 : 6"
```
`hover` isn't defined anywhere, so this is throwing an error.

mmwoodfo commented on wordle-web/components/keyboard.vue:
```diff
@@ -74,8 +87,22 @@ export default class KeyBoard extends Vue {
+           let stringWord: string="";
+           for(let i=0;i<5;i++){
+               stringWord += word.letters[i].char;
+           }
```
```suggestion
           let stringWord: string= word.letters.map(x => x.char).join('');
```
Another way you could go about converting this array to a string.

mmwoodfo reviewed (commented):
Make sure you guys run lint in your codebase, there were a lot of files that could have been auto-fixed as well as 2 typescript errors that could have been detected and resolved.

![image](https://user-images.githubusercontent.com/43832670/166854971-bef7b86c-d092-4e8e-9170-3fdbce6f4f1b.png)

mmwoodfo commented:
It wasn't clear for this assignment, and that is just Grant and I's fault, but for Assignment 3 and onward, make the PR against the class's Assignment's branch rather than the class main.

There should be explicit instructions in the Assignment 3 description. Good job making an Assignment 2 branch on your end though!

mmwoodfo commented on wordle-web/pages/about.vue:
```diff
@@ -0,0 +1,33 @@
+      how to effectively use Vuetify, Nuxt, GitHub, and Azure. In particular,
+      over the course of 10 weeks, we will build several things, but the first
+      is a Wordle-inspired game. Meg is really great and everyone likes how she
+      gives extra-credit for people saying "Meg is really great."
```
I think this counts as bribery 🤔 lol

mmwoodfo reviewed (commented):
I'll be reviewing from this commit: 
https://github.com/IntelliTect-Samples/EWU-CSCD379-2022-Spring/pull/39/commits/2960202d4140d1cb75fb4fd427c9b2304fbb4ad7
